backend/notifications/models.py

```diff
@@ -0,0 +1,184 @@
+    identifier = models.CharField(
+        _("identifier"),
+        max_length=200,
+        choices=Identifier.choices,
+    )
+
+    reply_to = models.EmailField(_("reply to"), blank=True, null=True)
+    subject = models.TextField(_("subject"))
+    preview_text = models.TextField(_("preview text"), blank=True)
+    body = models.TextField(_("body"))
+    cc_addresses = models.JSONField(_("cc addresses"), default=list, blank=True)
+    bcc_addresses = models.JSONField(_("bcc addresses"), default=list, blank=True)
+
+    objects = EmailTemplateQuerySet().as_manager()
+
+    def __str__(self):
+        if self.identifier == self.Identifier.custom:
+            return f"EmailTemplate {self.name} ({self.conference})"
+
+        return f"EmailTemplate {self.identifier} ({self.conference})"
+
+    def send_email(
+        self,
+        *,
+        recipient: User | None = None,
+        recipient_email: str | None = None,
+        placeholders: dict = None,
+    ):
+        if not recipient and not recipient_email:
+            raise ValueError("Either recipient or recipient_email must be provided")
+
+        from notifications.tasks import send_pending_emails
+
+        recipient_email = recipient_email or recipient.email
+        placeholders = placeholders or {}
+        processed_subject = render_template_from_string(
+            self.subject, placeholders
+        ).replace("\n", "<br/>")
+        processed_preview_text = render_template_from_string(
+            self.preview_text, placeholders
+        ).replace("\n", "<br/>")
+        processed_body = mark_safe(
+            render_template_from_string(
+                self.body,
+                placeholders,
+            ).replace("\n", "<br/>")
+        )
+
+        html_body = render_to_string(
+            "notifications/email-template.html",
+            {
+                "subject": processed_subject,
+                "preview_text": processed_preview_text,
+                "body": processed_body,
+            },
+        )
+
+        SentEmail.objects.create(
+            email_template=self,
+            conference=self.conference,
+            recipient=recipient,
+            recipient_email=recipient_email,
+            placeholders=placeholders,
+            subject=processed_subject,
+            preview_text=processed_preview_text,
+            body=html_body,
+            reply_to=self.reply_to,
+            cc_addresses=self.cc_addresses,
+            bcc_addresses=self.bcc_addresses,
+        )
+
+        transaction.on_commit(lambda: send_pending_emails.delay())
+
+    @property
+    def is_custom(self):
+        return self.identifier == self.Identifier.custom
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["identifier", "conference"],
+                condition=~Q(identifier="custom"),
+                name="unique_non_custom_identifier_conference",
+            )
+        ]
+
+
+class SentEmail(TimeStampedModel):
```

Comment:
verify this model can be used for logging purposes but also to cover all of future needs to know "did we notify this user already?"

i.e. in grants we want to automate sending reminders to accept the grant, we should be able to use this model to answer "Has it been more than 3 days since we sent the reminder to accept their grants?" and "How many reminders have we sent to this user so far?"

Reply:
my idea: 

```
SentEmail.objects.for_conference(conference).for_template(ACCEPT_GRANT).for_recipient(grant_user)
```

this would give us all accept grant emails sent to this user for this conference

```
SentEmail.objects.for_conference(conference).for_template(ACCEPT_GRANT).for_recipient(grant_user).order_by('-sent_at').first()
```

this would get us the last email sent

```
SentEmail.objects.for_conference(conference).for_template(ACCEPT_GRANT).for_recipient(grant_user).count()
```

gives us the count of accept grant sent so far

Reply:
worth checking the behaviour of pending/sent

what if an email is suck and never sent? would be forever in `pending` state with `sent_at=None`